py/src/LinearOptimizer.py

```diff
@@ -42,12 +37,29 @@ def intercepting_traj(self, p0, v0, pf, vf, T):
             print("Intercepting trajectory: optimization failed")
 
         u_values = u_sol.vector_values(u_sol.get_segment_times())
-        return result.is_success(), u_values 
+        return result.is_success(), u_values
     
     # TODO: minimum time trajectory
-    # given initial state, final state
-    #def min_time_traj(self, x0, v0, xf, vf):
-    #   return False
+    # Reach a certain position in minimum time, regardless of anything else.
+    def min_time_traj(self, p0, v0, pf, vf, xlim=None, ylim=None):
+        T = 1
+        x0 = np.concatenate((p0, v0), axis=0)
+        prog = DirectTranscription(self.sys, self.sys.CreateDefaultContext(), int(T/self.params.dt))
+        prog.AddBoundingBoxConstraint(x0, x0, prog.initial_state()) # initial states
+        prog.AddBoundingBoxConstraint(pf, pf, prog.final_state()[0:2])
+
+        self.add_input_limits(prog)
+        self.add_arena_limits(prog)
+
+        prog.AddRunningCost(prog.input()[0]*prog.input()[0])
```

Comment:
My understanding is that `prog.input()[0]` is the x-vel and `prog.input()[1]` is the y-vel. Should we do 
`prog.input()[0]*prog.input()[0]+prog.input()[1]*prog.input()[1]`

Not that this is important since we want to do minimum time later

Reply:
Agreed about the cost function formulation. But also agree that this is mainly a placeholder for the future implementation.